examples/crowdfunding/test/failing-tests.js

```diff
@@ -0,0 +1,437 @@
+  });
+
+  it('should fail donation if timestamp is after endDate', () => {
+    updateAndOptIn();
+    // set timestamp to after of endDate
+    runtime.setRoundAndTimestamp(5, endDate.getSeconds() + 100);
+
+    assert.throws(() => runtime.executeTx(donateTxGroup), rejectMsg);
+  });
+
+  it('should fail donation if timestamp is before of beginDate', () => {
+    updateAndOptIn();
+    // set timestamp to out of endDate
+    runtime.setRoundAndTimestamp(5, beginDate.getSeconds() - 100);
+
+    assert.throws(() => runtime.executeTx(donateTxGroup), rejectMsg);
+  });
+
+  it('should fail if goal is met, and donor tries to reclaim funds', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    runtime.setRoundAndTimestamp(5, endDate.getTime() + 100); // end date is passed
+
+    // donor should be able to reclaim if goal is NOT met
+    appArgs = [stringToBytes('reclaim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs,
+        accounts: [escrow.address] //  AppAccounts
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: donor.address,
+        amountMicroAlgos: 300000,
+        lsig: lsig,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), rejectMsg);
+  });
+
+  it('should fail if goal is not met, donor tries to reclaim funds before fund close date', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+
+    // donor should be able to reclaim if goal is NOT met
+    appArgs = [stringToBytes('reclaim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs,
+        accounts: [escrow.address] //  AppAccounts
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: donor.address,
+        amountMicroAlgos: 300000,
+        lsig: lsig,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), rejectMsg);
+  });
+
+  it('should fail if creator tries to claim funds before fund end date', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
```

Comment:
Same as above. If we won't be removing donate then please add a comment. Also, if we are doing this twice in multiple tests, not sure but it would probably affect the speed of tests as well.

Reply:
we can do one more thing increase donation amount to greater than goal.